lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py

```diff
@@ -801,130 +807,41 @@ def create_or_update_load_balancer(self, param):
-
-    if load_balancer.load_balancing_rules:
-        result['load_balancing_rules'] = [dict(
-            id=_.id,
-            name=_.name,
-            protocol=_.protocol,
-            frontend_ip_configuration_id=_.frontend_ip_configuration.id,
-            backend_address_pool_id=_.backend_address_pool.id,
-            probe_id=_.probe.id,
-            load_distribution=_.load_distribution,
-            frontend_port=_.frontend_port,
-            backend_port=_.backend_port,
-            idle_timeout_in_minutes=_.idle_timeout_in_minutes,
-            enable_floating_ip=_.enable_floating_ip,
-            provisioning_state=_.provisioning_state,
-            etag=_.etag
-        ) for _ in load_balancer.load_balancing_rules]
-
-    if load_balancer.probes:
-        result['probes'] = [dict(
-            id=_.id,
-            name=_.name,
-            protocol=_.protocol,
-            port=_.port,
-            interval_in_seconds=_.interval_in_seconds,
-            number_of_probes=_.number_of_probes,
-            request_path=_.request_path,
-            provisioning_state=_.provisioning_state
-        ) for _ in load_balancer.probes]
-
-    if load_balancer.inbound_nat_rules:
-        result['inbound_nat_rules'] = [dict(
-            id=_.id,
-            name=_.name,
-            frontend_ip_configuration_id=_.frontend_ip_configuration.id,
-            protocol=_.protocol,
-            frontend_port=_.frontend_port,
-            backend_port=_.backend_port,
-            idle_timeout_in_minutes=_.idle_timeout_in_minutes,
-            enable_floating_point_ip=_.enable_floating_point_ip if hasattr(_, 'enable_floating_point_ip') else False,
-            provisioning_state=_.provisioning_state,
-            etag=_.etag
-        ) for _ in load_balancer.inbound_nat_rules]
-
-    if load_balancer.inbound_nat_pools:
-        result['inbound_nat_pools'] = [dict(
-            id=_.id,
-            name=_.name,
-            frontend_ip_configuration_id=_.frontend_ip_configuration.id,
-            protocol=_.protocol,
-            frontend_port_range_start=_.frontend_port_range_start,
-            frontend_port_range_end=_.frontend_port_range_end,
-            backend_port=_.backend_port,
-            provisioning_state=_.provisioning_state,
-            etag=_.etag
-        ) for _ in load_balancer.inbound_nat_pools]
-
-    if load_balancer.outbound_nat_rules:
-        result['outbound_nat_rules'] = [dict(
-            id=_.id,
-            name=_.name,
-            allocated_outbound_ports=_.allocated_outbound_ports,
-            frontend_ip_configuration_id=_.frontend_ip_configuration.id,
-            backend_address_pool=_.backend_address_pool.id,
-            provisioning_state=_.provisioning_state,
-            etag=_.etag
-        ) for _ in load_balancer.outbound_nat_rules]
-
-    return result
+def default_compare(new, old, path):
+    if new is None:
+        return True
+    elif isinstance(new, dict):
+        if not isinstance(old, dict):
```

Comment:
if new is dict, and old is None, that means previously user didn't set some  properties(None), now, user want to update it(dict), why return False?

Reply:
so that correct, returning False will trigger update.

Reply:
ok. so true if compare equals, false if not. Seems like equal?

Reply:
yes, exactly